mucommander-os-api/src/main/java/com/mucommander/desktop/DefaultDesktopAdapter.java

```diff
@@ -155,4 +156,46 @@ public boolean isApplication(AbstractFile file) {
     public ActionShortcuts getActionShortcuts() {
         return new ActionShortcuts();
     }
+
+    @Override
+    public boolean setIconBadgeNumber(int number) {
+        boolean opSuccessful = false;
+        if (Taskbar.isTaskbarSupported() &&
+                Taskbar.getTaskbar().isSupported(Taskbar.Feature.ICON_BADGE_TEXT)) {
+            if (number >= 0) {
+                Taskbar.getTaskbar().setIconBadge(Integer.toString(number));
+            } else {
+                Taskbar.getTaskbar().setIconBadge(null);
+            }
+            opSuccessful = true;
+        }
+        return opSuccessful;
+    }
+
+    @Override
+    public boolean setIconProgress(int progress) {
+        boolean opSuccessful = false;
+        if (Taskbar.isTaskbarSupported() &&
+                Taskbar.getTaskbar().isSupported(Taskbar.Feature.PROGRESS_VALUE)) {
+            if (progress >= 0 && progress <= 100) {
+                Taskbar.getTaskbar().setProgressValue(progress);
+            } else {
+                Taskbar.getTaskbar().setProgressValue(-1);
+            }
+            opSuccessful = true;
+        }
+        return opSuccessful;
+    }
+
+    @Override
+    public boolean requestUserAttention() {
+        boolean opSuccessful = false;
+        if (Taskbar.isTaskbarSupported() &&
+                Taskbar.getTaskbar().isSupported(Taskbar.Feature.USER_ATTENTION)) {
+            // TODO check if we need to "unrequest" attention
```

Comment:
yeah and also we may want to set the window, in case there are multiple main frames (but again, I'm not sure it's that important - I don't find much use for using more than one frame)

Reply:
> yeah and also we may want to set the window, in case there are multiple main frames

if I get this right, then there's no way to specify main frame while invoking Taskbar.getTaskbar().requestUserAttention. On macOS there's one dock icon for multiple main frames, don't remember how it is on Windows or Linux.

Reply:
right, but I saw requestWindowUserAttention - not sure if those two methods can go well together

Reply:
ah yes, however, job listeners don't have a clue what main frame executed them so, I would try to go with a simple `requestUserAttention` and see if it is OK for users?

Reply:
> > @ahadas / @Andreas0602 or anyone - do you have account on bugs.openjdk.org ...
> 
> No, I haven't.

Me neither